boltstub_rs/src/values/bolt_struct/unknown_type.rs

```diff
@@ -0,0 +1,140 @@
+            i,
+            "UT",
+            config,
+        )?;
+        check_last_json_field(&mut fields, i, "UT")?;
+        if extra.contains_key("message") {
+            let PackStreamValue::String(ref message) = extra["message"] else {
+                return Err(ParseError::new("Expected message in extra to be string."));
+            };
+            Ok(Self {
+                name,
+                minimum_protocol_major,
+                minimum_protocol_minor,
+                message: Some(message.clone()),
+            })
+        } else {
+            Ok(Self {
+                name,
+                minimum_protocol_major,
+                minimum_protocol_minor,
+                message: None,
+            })
+        }
+    }
+
+    pub(crate) fn into_struct(self) -> PackStreamStruct {
+        let mut extra_map = IndexMap::with_capacity(1);
+        if self.message.is_some() {
+            extra_map.insert(
+                String::from("message"),
+                PackStreamValue::String(self.message.unwrap()),
+            );
+        }
+        let fields = vec![
+            PackStreamValue::String(self.name),
+            PackStreamValue::Integer(self.minimum_protocol_major),
+            PackStreamValue::Integer(self.minimum_protocol_minor),
+            PackStreamValue::Dict(extra_map),
+        ];
+        PackStreamStruct {
+            tag: TAG_UNKNOWN_TYPE,
+            fields,
+        }
+    }
+}
+#[derive(Debug, Clone)]
+pub(super) struct BoltUnknownType(pub(super) JoltUnknownType);
+impl BoltUnknownType {
+    pub(super) fn from_struct(s: &PackStreamStruct, jolt_version: JoltVersion) -> Option<Self> {
+        if !matches!(jolt_version, JoltVersion::V3) || s.tag != TAG_UNKNOWN_TYPE {
+            return None;
+        }
+        let mut fields = s.fields.iter();
+        let name: &str = next_pack_stream_field(&mut fields)?;
+        let minimum_protocol_major: i64 = next_pack_stream_field(&mut fields)?;
+        let minimum_protocol_minor: i64 = next_pack_stream_field(&mut fields)?;
+        let extra: &IndexMap<String, PackStreamValue> = next_pack_stream_field(&mut fields)?;
+        let PackStreamValue::String(ref message) = extra["message"] else {
+            return None;
+        };
+        Some(Self(JoltUnknownType {
+            name: name.to_string(),
+            minimum_protocol_major,
+            minimum_protocol_minor,
+            message: Some(message.clone()),
+        }))
+    }
+
+    pub(super) fn jolt_fmt(&self, _jolt_version: JoltVersion) -> impl Display + '_ {
+        struct JoltFormatter<'a> {
+            this: &'a JoltUnknownType,
+        }
+
+        impl Display for JoltFormatter<'_> {
+            fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+                f.write_str(r#"{"UT": [""#)?;
+                f.write_str(&self.this.name)?;
+
+                f.write_str(r#"", ""#)?;
+                f.write_str(&self.this.minimum_protocol_major.to_string())?;
```

Comment:
```suggestion
                Display::fmt(&self.this.minimum_protocol_major, f)?;
```
I reckon this saves at least on String allocation (not tested).